integration-tests/docker/docker-compose.query-retry-test.yml

```diff
@@ -0,0 +1,128 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: "2.2"
```

Comment:
If the only difference between this one and the base yml is the new druid-historical-for-query-retry-test, you can just have the new druid-historical-for-query-retry-test in a yml and run two yml when you want to start druid (docker-compose -f ${DOCKERDIR}/docker-compose.yml -f ${DOCKERDIR}/docker-compose.query-retry-test.yml  up -d

Reply:
This is to avoid repeating things like druid-zookeeper-kafka etc.

Reply:
This docker compose file doesn't include middleManager to save memory. I would like to exclude overlord as well, seems like I need a way to avoid the check in `SuiteListener` that waits for the overlord to be ready. I haven't looked at it closely yet.

Reply:
I see. My only concern is duplicated code and the maintainability.